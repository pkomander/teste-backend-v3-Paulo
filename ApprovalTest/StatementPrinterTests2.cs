using Aiko.Dto;
using Aiko.Repository.Context;
using Aiko.Repository.Services.Interface;
using Aiko.Repository.Services.Repository;
using Aiko_teste.Util;
using ApprovalTests;
using ApprovalTests.Reporters;
using AutoMapper;
using Azure;
using FluentResults;
using Microsoft.EntityFrameworkCore;
using Moq;

namespace ApprovalTest
{
    public class StatementPrinterTests2
    {
        private readonly IExtratoService _extratoService;
        private IUtil _util;
        private readonly DataContext _context;

        public StatementPrinterTests2(IUtil util)
        {
            var options = new DbContextOptionsBuilder<DataContext>()
            .UseSqlServer("server=localhost;database=AikoTeste;user=sa;password=#sa123456;TrustServerCertificate=true;")
            .Options;

            _context = new DataContext(options);

            // Criar mocks das dependências
            //var mockContext = new Mock<DataContext>();
            //var context = new DataContext(options);
            var mockMapper = new Mock<IMapper>();

            _extratoService = new ExtratoService(_context, mockMapper.Object);
            _util = util;
        }

        [Fact]
        [UseReporter(typeof(DiffReporter))]
        public async void TestXmlStatementExample()
        {
            var objectTeste = new ExtratoRequest
            {
                InvoiceId = 1,
                Custumer = null,
                FormatoArquivo = 1
            };

            var gerarTexto = await _extratoService.GerarExtrato(objectTeste);

            await _util.CriarArquivo(gerarTexto.Successes.FirstOrDefault().Message, "Extrato", objectTeste.FormatoArquivo);


            //Approvals.VerifyXml(gerarTexto.Successes.FirstOrDefault());
            //Assert.True(gerarTexto.IsSuccess);
        }
    }
}
