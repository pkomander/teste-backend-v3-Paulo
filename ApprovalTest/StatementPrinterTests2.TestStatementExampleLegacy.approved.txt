Success with Message='Statement for BigCo
  Hamlet: R$ 402,50 (55 seats)
  As You Like It: R$ 269,80 (35 seats)
  Othello: R$ 357,00 (40 seats)
Amount owed is R$ 1.029,30
You earned 47 credits
'